### PR TITLE
Multiple ACLineSegments inside a Line container

### PR DESCRIPTION
**Pull request by marcosmc**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
Not allowed multiple ACLineSegments inside a Line container.

**What is the new behavior (if this is a feature change)?**
In configurations with multiple ACLineSegment inside a Line container, a VoltageLevel without Substation is created for each ConnetivityNode whose container is the Line and all ACLineSegment are converted individually.

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 1]
```diff
@@ -380,13 +381,108 @@ private void convertACLineSegmentsToLines(Context context, Set<String> delayedBo
             if (c.valid()) {
                 String node = c.boundaryNode();
                 if (node != null && !context.config().convertBoundary()) {
-                    context.boundary().addAcLineSegmentAtNode(line, node);
-                    delayedBoundaryNodes.add(node);
+                    String lineContainerId = line.getId("Line");
+                    if (lineContainerId != null) {
+                        delayedLines.computeIfAbsent(lineContainerId, l -> new PropertyBags()).add(line);
```
Comment (zamarrenolm) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java:386]: Here we are recording ONLY `ACLineSegments` that are contained inside a `cim:Line` AND have one end at the boundary. We could have `ACLineSegments` inside the network, none of the two ends placed at the boundary.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 2]
```diff
@@ -380,13 +381,108 @@ private void convertACLineSegmentsToLines(Context context, Set<String> delayedBo
...
+                    ACLineSegmentConversion c = new ACLineSegmentConversion(line, context);
+                    context.boundary().addAcLineSegmentAtNode(line, c.boundaryNode());
+                    delayedBoundaryNodes.add(c.boundaryNode());
+                });
+                delayedLines.remove(lineContainerId, p);
+            }
+        });
+    }
+
+    private void convertLine(Context context, String lineId, PropertyBags lineSegments) {
```
Comment (zamarrenolm) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java:411]: I am not sure that we need the approach of delaying the conversion of `ACLineSegments` found in `cim:Line` containers. For me, they should be converted using the current existing considerations (could be located at boundaries, could be merged with others at boundary, ...). The only consideration that we must add is that one of the ends will be a fictitious voltage level, that we create as we need it. We should create a fictitious voltage level for every `ConnectivityNode` inside the cim:Line; all the `ACLineSegments` sharing the same connectivity node in the CGMES model will be mapped to IIDM Lines in the same fictitious voltage level.

**Review by zamarrenolm — CHANGES_REQUESTED** [review]

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 3]
```diff
@@ -370,25 +369,116 @@ private void addCimCharacteristics(Network network) {
...
         while (k.hasNext()) {
             PropertyBag line = k.next();
             if (LOG.isDebugEnabled()) {
                 LOG.debug(line.tabulateLocals("ACLineSegment"));
             }
             ACLineSegmentConversion c = new ACLineSegmentConversion(line, context);
+            if (c.valid()) {
+                String lineContainerId = line.getId("Line");
+                if (lineContainerId != null) {
+                    lineContainers.computeIfAbsent(lineContainerId, l -> new PropertyBags()).add(line);
```
Comment (zamarrenolm) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java:383]: After the latest changes, I still see no reason to delay the conversion of ACLS contained in a cim:Line. We just have to create the fictitious voltage levels and/or substations associated to them.

**Review by zamarrenolm — CHANGES_REQUESTED** [review]

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 4]
```diff
@@ -389,6 +393,66 @@ private void convertACLineSegmentsToLines(Context context, Set<String> delayedBo
...
+                .setTopologyKind(
+                        context.nodeBreaker()
+                                ? TopologyKind.NODE_BREAKER
+                                : TopologyKind.BUS_BREAKER)
+                .setLowVoltageLimit(lowVoltageLimit)
+                .setHighVoltageLimit(highVoltageLimit)
+                .setId(id)
+                .setName(name)
+                .setEnsureIdUnicity(context.config().isEnsureIdAliasUnicity())
+                .add();
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java:453]: I think you can put the Line container ID as property of the contained Voltage Level

**Review by MioRtia — APPROVED** [review]
One small remark but otherwise it is okay for me!

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java (hunk 4 above)
Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java:453]: Done. We also rearranged a little bit the code.

**Review by zamarrenolm — APPROVED** [review]

**Pull request merged**